src/Tribe/Aggregator/Tabs/New.php

```diff
@@ -158,11 +165,28 @@ public function handle_import_finalize( $data ) {
 			'warning',
 		);
 
-		if ( 'csv' === $data['origin'] ) {
-			$result = $record->csv_insert_posts( $data );
+		$result = $record->insert_posts( $data );
+
+		if ( is_wp_error( $result ) ) {
+			$this->messages[ 'error' ][] = $result->get_error_message();
+
+			Tribe__Admin__Notices::instance()->register(
```

Comment:
Can we do it on a single line?
```php
Tribe__Admin__Notices::instance()->register( 'aggregator-import-failed', array( $this, 'render_notice_import_failed' ), 'type=error' )
```

I will add the `tribe_admin_notice()` public function.